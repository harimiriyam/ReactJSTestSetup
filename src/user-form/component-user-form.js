import React from 'react'
import { PropTypes } from 'prop-types'


export const UserDataForm = ({ name, contact, dispatch }) => {
	
	let _name, _contact;

	return (
			<div className='form-container'>			
			<form onSubmit={() => { 
				dispatch({
					type: 'ADD_USER', 
					name: _name.value,
					contact: _contact.value
				})
                _name.value = "";
                _contact.value = "";
                }} className='input-form'>
			<div>
				<label>User Name :</label>
				<input type='text' id='userName' ref={(input) => { _name = input; }}/>
			</div>
			<div>
			<label>Contact :</label>
			<input type='text' id='contact' ref={(input) => { _contact = input; }} />
			</div>
			<button className="formSubmit">Submit</button>
			</form></div>
			
		)
}

UserDataForm.propTypes = {
	name: PropTypes.string,
	contact: PropTypes.number
}