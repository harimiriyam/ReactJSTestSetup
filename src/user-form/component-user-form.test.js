import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { UserDataForm } from './component-user-form';

it('should correctly render user form', () => {
  const result = renderer.create(
      <UserDataForm />
  );
  expect(result.toJSON()).toMatchSnapshot();
});