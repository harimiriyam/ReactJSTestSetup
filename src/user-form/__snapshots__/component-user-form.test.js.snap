// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render user form 1`] = `
<div
  className="form-container"
>
  <form
    className="input-form"
    onSubmit={[Function]}
  >
    <div>
      <label>
        User Name :
      </label>
      <input
        id="userName"
        type="text"
      />
    </div>
    <div>
      <label>
        Contact :
      </label>
      <input
        id="contact"
        type="text"
      />
    </div>
    <button
      className="formSubmit"
    >
      Submit
    </button>
  </form>
</div>
`;
